Develop algorithms for gene deletion strategies for growth-coupled production

MetNetComp opened the pull request:
Dear Administrator,

TrimGdel appropriately considers GPR rules and derives the minimal and maximal gene deletion strategies for growth-coupled production.

T. Tamura, "Trimming Gene Deletion Strategies for Growth-Coupled Production in Constraint-Based Metabolic Networks: TrimGdel," in IEEE/ACM Transactions on Computational Biology and Bioinformatics, vol. 20, no. 2, pp. 1540-1549, 2023.

Best regards,
Takeyuki Tamura,
Kyoto University, Japan.

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [ ] with [X] to check the box)*

rmtfleming reviewed (commented):
remove src/design/TrimGdel/e_coli_core.mat because that is in COBRA.models

Also ALL functions should have complete headers. See printFunctionIO

rmtfleming commented:
Also, please add test function to 
https://github.com/opencobra/cobratoolbox/tree/master/test/verifiedTests
Also, is there a tutorial to help others understand how to use your code?
https://github.com/opencobra/COBRA.tutorials
See
https://opencobra.github.io/cobratoolbox/stable/contributing.html

MetNetComp commented:
Dear Ronan

Thank you for your review.
I removed e_coli_core; added complete headers to all functions;
added the test function to /test/verifiedTests/.

Also, I tried to create a pull request for ~/COBRA.tutorials,
but it seems that I do not have permission.

Best regards,
Takeyuki

MetNetComp commented:
Also, I would like to add iML1515.mat and iMM904.mat to ~/COBRA.models/mat/.

MetNetComp commented:
I would appreciate it if you could let me know of any additional points that need modification.

farid-zare commented on src/design/TrimGdel/testTrimGdel.m:
Hi,
As you have already added this test function in the test folder, I believe you could please remove it from the src code folder.
Thanks

MetNetComp replied:
Thanks. I have removed src/design/TrimGdel/testTrimGdel.m.
